match-standards-serum-mix01.Rmd

```diff
@@ -1195,108 +706,107 @@ The table below lists all features that were matched to one of the adducts of
+In addition we match (**all**) the MS2 spectra for the matched features against
+all spectra from HMDB or MassBank identifying reference spectra with a
+similarity larger than 0.7. The results (if any spectra matched) are shown in
+the two following tables.
+
+```{r, echo = FALSE, message = FALSE, results = "asis"}
+perform_match(std_ms2, hmdb, sv = c("rtime", "target_name", "score"),
+              name = "target_name", param = csp)
 ```
 
-```{r}
-mbank_match <- matchSpectra(
-    std_ms2, mbank,
-    param = CompareSpectraParam(ppm = 40, requirePrecursor = FALSE))
-mbank_match
+```{r, echo = FALSE, message = FALSE, results = "asis"}
+perform_match(std_ms2, mbank, sv = c("rtime", "target_name", "score"),
+              name = "target_name", param = csp)
 ```
 
-```{r, echo = FALSE, results = "asis", eval = length(whichQuery(mbank_match)) > 0}
-st <- match_table(mbank_match, c("rtime", "target_compound_name",
-                                 "target_exactmass", "score")) |>
-summarize_match_table("target_compound_name")
-pandoc.table(
-    st, style = "rmarkdown", split.table = Inf,
-    caption = paste0("Experimental MS2 spectra with best match to MS2",
-                     " spectra of different compounds. The best match per ",
-                     "compound is reported. Rows are ordered by similarity."))
+For completeness, we also perform a neutral loss spectra comparison with HMDB
+and MassBank (requiring a high similarity).
+
+```{r, echo = FALSE, message = FALSE, results = "asis", eval = ALL_NL_MATCH}
+std_ms2_nl <- neutralLoss(std_ms2, PrecursorMzParam())
+perform_match(std_ms2_nl, hmdb_nl, sv = c("rtime", "target_name", "score"),
+              name = "target_name", param = csp_nl, similarity = 0.9)
 ```
 
-The feature with a retention time around 177 seconds (FT0752) seem to represent
-signal from `r std` (being only matched to `r std` and with a high score).
+```{r, echo = FALSE, message = FALSE, results = "asis", eval = ALL_NL_MATCH}
+perform_match(std_ms2_nl, mbank_nl, similarity = 0.9,
+              sv = c("rtime", "target_name", "score"),
+              name = "target_name", param = csp_nl)
+```
 
 #### Summary
 
-- FT00870 (RT=176, `[M+H]+` ion): confidence **C**: the MS2 spectrum F07.S0424
-  matches only `r std`.
-- FT01667 and FT01307 inherit confidence **C** from FT00870.
-- Reference MS2: F23.S0633?
-- Also in this case we have spectra that match reference ones from Creatinine
-  with relatively high scores.
+- MS2 spectra for FT00578, FT00577 and FT00575 match spectra from creatinine.
+- FT00870 (RT=176, `[M+H]+` ion): confidence **C**: the MS2 spectrum F07.S0633
+  matches only `r std` with a low similarity.
+- FT01667 (RT=173.3, `[M+2Ma-H]+` ion): neutral loss spectrum of F07.S0621
+  matches reference spectrum with high simlarity. Confidence **B**?.
+- FT01307 inherit confidence from above.
+- Reference spectra: F07.S0621 (and F07.S0633?).
+
+
+TODO CONTINUE/REPLACE BELOW
```

Comment:
@andreavicini , I've updated all stuff above, please continue here replacing the old code with the new chunks defined above.